__tests__/Auth0Client.test.ts

```diff
@@ -2055,6 +2057,98 @@ describe('Auth0Client', () => {
     });
   });
 
+  describe('getTokenWithPopup()', () => {
+    const localSetup = async (clientOptions?: Partial<Auth0ClientOptions>) => {
+      const auth0 = setup(clientOptions);
+
+      setupMessageEventLister(mockWindow, { state: TEST_STATE });
+
+      mockFetch.mockResolvedValueOnce(
+        fetchResponse(true, {
+          id_token: TEST_ID_TOKEN,
+          refresh_token: TEST_REFRESH_TOKEN,
+          access_token: TEST_ACCESS_TOKEN,
+          expires_in: 86400
+        })
+      );
+
+      return auth0;
+    };
+
+    it('calls `loginWithPopup` with the correct default options', async () => {
+      const auth0 = await localSetup();
+      expect(await auth0.getTokenWithPopup()).toEqual(TEST_ACCESS_TOKEN);
+    });
+
+    it('respects customized scopes', async () => {
+      const auth0 = await localSetup({
+        advancedOptions: {
+          defaultScope: 'email'
+        },
+        scope: 'read:email'
+      });
+
+      const config = {
+        popup: {
+          location: {
+            href: ''
+          },
+          close: jest.fn()
+        }
+      };
+
+      expect(await auth0.getTokenWithPopup({}, config)).toEqual(
+        TEST_ACCESS_TOKEN
+      );
+
+      expect(config.popup.location.href).toMatch(
+        /openid%20email%20read%3Aemail/
+      );
+    });
+
+    it('passes custom login options', async () => {
+      const auth0 = await localSetup();
+
+      const loginOptions = {
+        audience: 'other-audience',
+        screen_hint: 'signup'
+      };
+
+      const config = {
+        popup: {
+          location: {
+            href: ''
+          },
+          close: jest.fn()
+        }
+      };
+
+      await auth0.getTokenWithPopup(loginOptions, config);
+
+      expect(config.popup.location.href).toMatch(/other-audience/);
+      expect(config.popup.location.href).toMatch(/screen_hint/);
```

Comment:
`getTokenWithPopup` still does some of its own logic to pass in the right login options, so I think this test is valuable.